business/workloads.go

```diff
@@ -273,23 +279,25 @@ func FilterUniqueIstioReferences(refs []*models.IstioValidationKey) []*models.Is
 
 // GetWorkload is the API handler to fetch details of a specific workload.
 // If includeServices is set true, the Workload will fetch all services related
-func (in *WorkloadService) GetWorkload(ctx context.Context, namespace string, workloadName string, workloadType string, includeServices bool) (*models.Workload, error) {
+func (in *WorkloadService) GetWorkload(ctx context.Context, criteria WorkloadCriteria) (*models.Workload, error) {
```

Comment:
Args are different between the exported funcs `GetWorkload` and `GetWorkloadList`.

It is misleading when you want to use the funcs, because you won't know which fields you need to really fill in each case without looking at the code. So, I would prefer that each func has it's own args type.

I think this same comment applies to the other files...

Not a blocker.